test_bst.py

```diff
@@ -266,3 +284,132 @@ def test_empty_tree_traversals(empty_bst):
+
+
+def test_delete_middle(bst_root_fifty):
+    bst_root_fifty.delete(25)
+    # import pdb; pdb.set_trace()
+    assert bst_root_fifty.root.value == 50
+    assert bst_root_fifty.root.left_child.value == 35 
+    assert bst_root_fifty.root.left_child.left_child is None
+    assert bst_root_fifty.root.left_child.right_child.value == 35
+    assert bst_root_fifty.root.left_child.right_child.parent.value == 12
+    assert not bst_root_fifty.contains(25)
+
+
+def test_delete_root(bst_root_fifty):
+    bst_root_fifty.delete(50)
+    root = bst_root_fifty.root
+    assert root.value == 25
+    assert root.right_child.value == 100
+    assert root.left_child.value == 12
+    assert root.right_child.right_child.value == 35
+    assert root.right_child.right_child.value == 150
+    assert root.right_child.left_child.value == 75
+# def test_delete_node_both_children(bst_root_fifty):
+#     bst_root_fifty.delete(100)
+#     assert not bst_root_fifty.contains(100)
+#     assert bst_root_fifty.root.right_child.left_child.value == 75
+#     assert bst_root_fifty.root.right_child.value == 150
+
+
+# def test_delete_node_both_children_root(bst_root_fifty):
+#     bst_root_fifty.delete(50)
+#     assert bst_root_fifty.root.value == 100
+
+
+# def test_delete_parent(bst_root_fifty):
+#     bst_root_fifty.insert(30)
+#     bst_root_fifty.insert(37)
+#     bst_root_fifty.delete(35)
+#     root_grandchild = bst_root_fifty.root.left_child.right_child
+#     assert root_grandchild.value == 37
+#     assert root_grandchild.left_child.value == 30
+
+
+# def test_delete_grampa(bst_root_fifty):
+#     bst_root_fifty.insert(6)
+#     bst_root_fifty.insert(13)
+#     bst_root_fifty.delete(25)
+#     root_left_child = bst_root_fifty.root.left_child
+#     assert root_left_child.value == 35
+#     assert root_left_child.left_child.value == 12
+
+
+# def test_delete_parent_one_child(bst_root_fifty):
+#     bst_root_fifty.insert(6)
+#     bst_root_fifty.delete(12)
+#     root_left_child = bst_root_fifty.root.left_child
+#     assert root_left_child.value == 25
+#     assert root_left_child.left_child.value == 6
+
+
+# def test_delete_parent_one_child_right(bst_root_fifty):
+#     bst_root_fifty.insert(27)
+#     bst_root_fifty.delete(35)
+#     root_left_child = bst_root_fifty.root.left_child
+#     assert root_left_child.value == 25
+#     assert root_left_child.right_child.value == 27
+
+
+# def test_delete_parent_two_children(bst_root_fifty):
+#     bst_root_fifty.insert(6)
+#     bst_root_fifty.insert(13)
+#     bst_root_fifty.delete(12)
+#     root_left_child = bst_root_fifty.root.left_child
+#     assert root_left_child.value == 25
+#     assert root_left_child.left_child.value == 13
+#     assert root_left_child.left_child.left_child.value == 6
+
+
+# def test_two_node_bst(empty_bst):
+#     empty_bst.insert(100)
+#     empty_bst.insert(75)
+#     empty_bst.insert(150)
+#     empty_bst.delete(100)
+#     assert empty_bst.root.value == 150
```

Comment:
If you aren't using these, you should not have these in here. At least not without documenting it really well so your readers know why they aren't being included.